### PR TITLE
[PhpUnitBridge][VarDumper] Fix color detection

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Porting #53707 to VarDumper and tweaking it a bit.

/cc @theofidry FYI

### CODE REVIEW

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php [hunk 1]
```diff
@@ -403,27 +403,58 @@ private static function hasColorSupport()
...
+        ) {
+            return true;
+        }
+
+        if ('dumb' === $term = (string) getenv('TERM')) {
+            return false;
+        }
+
+        // See https://github.com/chalk/supports-color/blob/d4f413efaf8da045c5ab440ed418ef02dbb28bf1/index.js#L157
+        return preg_match('/^((screen|xterm|vt100|vt220|putty|rxvt|ansi|cygwin|linux).*)|(.*-256(color)?(-bce)?)$/', $term);
```
Comment (theofidry) [inline src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php:430]: shouldn't it be `@\`? (not sure if the `pcre` extension is declared in the `composer.json` either

Comment (nicolas-grekas) [reply src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php:430]: why would it? preg_match doesn't trigger notices unless the regexp is badly formatted, which it isn't
about ext-pcre, we never list it as a requirement - it's one of the builtin extensions.

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php [hunk 2]
```diff
@@ -403,27 +403,58 @@ private static function hasColorSupport()
             return false;
         }
 
-        if ('Hyper' === getenv('TERM_PROGRAM')) {
+        if (!self::isTty()) {
             return true;
```
Comment (stof) [inline src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php:407]: this is broken. For not a TTY, it should says `false`

File: src/Symfony/Component/VarDumper/Dumper/CliDumper.php [hunk 3]
```diff
@@ -610,19 +610,30 @@ private function hasColorSupport($stream): bool
             return false;
         }
 
-        if ('Hyper' === getenv('TERM_PROGRAM')) {
+        // Detect msysgit/mingw and assume this is a tty because detection
+        // does not work correctly, see https://github.com/composer/composer/issues/9690
+        if (!@stream_isatty($stream) && !\in_array(strtoupper((string) getenv('MSYSTEM')), ['MINGW32', 'MINGW64'], true)) {
             return true;
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Dumper/CliDumper.php:616]: The logic in this method does not match the logic in the console component. A stream that is not a TTY must not report supporting colors.

**Pull request merged**